moveit_core/planning_scene/src/planning_scene.cpp

```diff
@@ -1807,25 +1790,31 @@ bool PlanningScene::processCollisionObjectMove(const moveit_msgs::CollisionObjec
       ROS_WARN_NAMED(LOGNAME, "Move operation for object '%s' ignores the geometry specified in the message.",
                      object.id.c_str());
 
-    const Eigen::Isometry3d& t = getTransforms().getTransform(object.header.frame_id);
+    const Eigen::Isometry3d& world_to_object_header_transform = getTransforms().getTransform(object.header.frame_id);
+    Eigen::Isometry3d object_header_to_pose_transform;  // TODO(felixvd): Should this assume a neutral quaternion if
+                                                        // all values are 0?
+    PlanningScene::poseMsgToEigen(object.pose, object_header_to_pose_transform);
+
+    const Eigen::Isometry3d& object_frame_transform =
+        world_to_object_header_transform * object_header_to_pose_transform;
     EigenSTL::vector_Isometry3d new_poses;
```

Comment:
If you want to optimize this a tiny bit, you can call .reserve on the vector and reserve the number of primitive_poses + mesh_poses + planes_poses.

Reply:
Added